api/src/integration/java/org/pmiops/workbench/FireCloudIntegrationTest.java

```diff
@@ -29,46 +47,131 @@
+
+    return new FireCloudServiceImpl(
         Providers.of(config),
         Providers.of(profileApi),
         Providers.of(billingApi),
-        // All of us groups api
-        Providers.of(groupsApi),
-        // End user groups api.
-        Providers.of(groupsApi),
+        Providers.of(allOfUsGroupsApi),
+        Providers.of(endUserGroupsApi),
         Providers.of(nihApi),
         Providers.of(workspacesApi),
-        Providers.of(new FireCloudConfig().statusApi(config)),
-        new FirecloudRetryHandler(new NoBackOffPolicy())
+        Providers.of(new StatusApi(apiClient)),
+        new FirecloudRetryHandler(new NoBackOffPolicy()),
+        serviceAccounts,
+        Providers.of(fireCloudAdminCredential)
     );
   }
 
-  @Test
-  public void testStatus() {
-    assertThat(fireCloudService.getFirecloudStatus()).isTrue();
-  }
-
-  private static WorkbenchConfig createConfig() throws Exception {
-    String testConfig = Resources.toString(Resources.getResource("config_prod.json"), Charset.defaultCharset());
+  private WorkbenchConfig loadConfig(String filename) throws Exception {
+    String testConfig = Resources.toString(Resources.getResource(filename), Charset.defaultCharset());
     WorkbenchConfig workbenchConfig = new Gson().fromJson(testConfig, WorkbenchConfig.class);
     workbenchConfig.firecloud.debugEndpoints = true;
     return workbenchConfig;
   }
 
+  private FireCloudService getTestService() throws Exception {
+    return createService(loadConfig("config_test.json"));
+  }
+
+  private FireCloudService getProdService() throws Exception {
+    return createService(loadConfig("config_prod.json"));
+  }
+
+  @Test
+  public void testStatusProd() throws Exception {
+    assertThat(getProdService().getFirecloudStatus()).isTrue();
+  }
+
+  @Test
+  public void testStatusDev() throws Exception {
+    assertThat(getTestService().getFirecloudStatus()).isTrue();
+  }
+
+  /**
+   * Ensure swe can successfully use delegation of authority to make FireCloud API calls on behalf
+   * of AoU users.
+   * <p>
+   * This test depends on there being an active account in FireCloud dev with the email address
+   * integration-test-user@fake-research-aou.org.
+   */
+  @Test
+  public void testImpersonatedProfileCall() throws Exception {
+    ApiClient apiClient = getTestService().getApiClientWithImpersonation(
+        "integration-test-user@fake-research-aou.org");
+
+    // Run the most basic API call against the /me/ endpoint.
+    ProfileApi profileApi = new ProfileApi(apiClient);
+    Me me = profileApi.me();
+    assertThat(me.getUserInfo().getUserEmail()).isEqualTo("integration-test-user@fake-research-aou.org");
+    assertThat(me.getUserInfo().getUserSubjectId()).isEqualTo("101727030557929965916");
+
+    // Run a test against a different FireCloud endpoint. This is important, because the /me/
+    // endpoint is accessible even by service accounts whose subject IDs haven't been whitelisted
```

Comment:
That's... slightly surprising.. I'm sure there's some history here. I don't suppose you know it off-hand? (probably not worth adding to the comments - just curious)